packages/navigator/src/browser/navigator-model.spec.ts

```diff
@@ -0,0 +1,304 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { enableJSDOM } from '@theia/core/lib/browser/test/jsdom';
+let disableJSDOM = enableJSDOM();
+
+import { Container } from 'inversify';
+import { Emitter, ILogger, Logger } from '@theia/core';
+import {
+    CompositeTreeNode, DefaultOpenerService, ExpandableTreeNode, LabelProvider, OpenerService,
+    Tree, TreeNode, TreeSelectionService, TreeExpansionService, TreeExpansionServiceImpl,
+    TreeNavigationService, TreeSearch
+} from '@theia/core/lib/browser';
+import { TreeSelectionServiceImpl } from '@theia/core/lib/browser/tree/tree-selection-impl';
+import { FileSystem, FileStat } from '@theia/filesystem/lib/common';
+import { FileSystemWatcher } from '@theia/filesystem/lib/browser/filesystem-watcher';
+import { FileSystemNode } from '@theia/filesystem/lib/node/node-filesystem';
+import { DirNode, FileChange, FileMoveEvent, FileTreeModel, FileStatNode } from '@theia/filesystem/lib/browser';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { FileNavigatorTree, WorkspaceNode, WorkspaceRootNode } from './navigator-tree';
+import { FileNavigatorModel } from './navigator-model';
+import URI from '@theia/core/lib/common/uri';
+import * as sinon from 'sinon';
+import * as chai from 'chai';
+const expect = chai.expect;
+
+disableJSDOM();
+
+let root: CompositeTreeNode;
+let workspaceRootFolder: DirNode;
+
+let childA: FileStatNode;
+let childB: FileStatNode;
+
+let homeFolder: DirNode;
+let childC: FileStatNode;
+
+let folderA: FileStat;
+let folderB: FileStat;
+
+// tslint:disable:no-any
+// tslint:disable:no-unused-expression
+const setup = () => {
```

Comment:
Can you add a comment to describe the file hierarchy you are emulating?

Reply:
Like:

```
- /home/rootFolder (root 1)
  - childA (file)
  - childB (file)
- /home (root 2)
  - ...
```

Reply:
yep i will do it in both spec files added